docs/source/cheat_sheet.rst

```diff
@@ -49,6 +49,15 @@ Built-in types
    if x is not None:
       print x
 
+   # There are 2 ways you can make mypy understand that
+   # an Optional value is not None
+   x = some_function()  # type: Optional[str]
+   # First way: with an if condition
+   if x is not None:
+       print(x.upper())
+   # Second way: with an assert statement
```

Comment:
These are not just two ways. They are semantically different. The cheat sheet should say something like:
```
# If it still may be None sometimes, use an if-statement
...
# If it can never be None (e.g. due to some invariants), use an assert
...
```